bot/cogs/grades_cog/grades_cog.py

```diff
@@ -50,20 +53,69 @@ def load_data(self):
+
+        # We have flexibility of the order and exact phrasing of arguments
+        for arg in args:
+            if arg in ("honors", "hon"):
+                honors = "honors"
+            elif arg in ("non-honors", "non-hon", "regular", "normal"):
+                honors = "non-honors"
+            elif arg == "all":
+                honors = "all"
+            elif arg.isdigit():
+                year = int(arg)
+            else:
+                error_title = "Unable to parse command"
+                error_message = 'Are you sure you used the proper format (ex: math-2060 2017 honors)?'
+
+        if len(args) > 2:
+            error_title = "Too many arguments"
+            error_message = 'Are you sure you used the proper format (ex: math-2060 2017 honors)?'
+
+        # Further processing of arguments
         if not self.grades_df.CourseId.str.contains(course).any():
+            error_title = "Course doesn't exist"
+            error_message = 'Are you sure you used the proper notation (ex: cpsc-2120)?'
+        elif self.grades_df[(self.grades_df.CourseId == course)].Year.max() < year:
+            error_title = "No data for year %s" % year
+            error_message = 'There is no grade data for your course for this year onward; please extend the range and try again.'
+        elif year < 2014:
+            error_title = "Year below minimum"
+            error_message = "The year you selected is below the minimum threshold (%d)." % MIN_YEAR
+
+        if error_title:
             embed = discord.Embed(title="Grades", color=Colors.Error)
-            result = 'Are you sure you used the proper notation (ex: cpsc-2120)?'
-            embed.add_field(name="ERROR: Course doesn't exist", value=result, inline=False)
-            embed.add_field(name=f'Help:', value= f'Run `{await self.bot.current_prefix(ctx)}grades list` to find what courses are available', inline=False)
+            embed.add_field(name="ERROR: " + error_title, value=error_message, inline=False)
+            embed.add_field(name=f'Help:',
+                            value=f'Run `{await self.bot.current_prefix(ctx)}help grades` for documentation or ' \
+                                  f'`{await self.bot.current_prefix(ctx)}grades list` to find out which ' \
+                                  'courses are available',
+                            inline=False)
             return await ctx.send(embed=embed)
 
+        # Default: no honors preference
         df = self.grades_df[(self.grades_df.CourseId == course) & (self.grades_df.Year >= year)]
-        title = df.Title.tolist()[0]
+
+        if honors == "honors":
+            df = self.grades_df[
+                (self.grades_df.CourseId == course) & (self.grades_df.Year >= year) & (self.grades_df.Honors == True)]
+        elif honors == "non-honors":
+            df = self.grades_df[(self.grades_df.CourseId == course) & (self.grades_df.Year >= year) & (
+                    self.grades_df.Honors != True)]
```

Comment:
what is the point of this?

Reply:
Yeah that could be better, it is supposed to filter for honors attribute. This is what I am working with currently:
```py
df = self.grades_df[(self.grades_df.CourseId == course) & (self.grades_df.Year >= year)]

if honors == 'honors':
    df = df[df.Honors == True]
elif honors == 'non-honors':
    df = df[df.Honors != True]
```